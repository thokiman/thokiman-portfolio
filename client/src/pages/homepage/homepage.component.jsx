import React, { lazy, Suspense } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import SpinnerLoading from '../../components/commons/spinner-loading/spinner-loading.component';
import {
  selectPersonalSummaryQuote,
  selectPersonalSummaryShortDescription,
} from '../../redux/about/about.selectors';
import { selectIsSideBarActive } from '../../redux/header/header.selectors';
import {
  HomepageAboutContentContainer,
  HomepageContainer,
  HomepageIsNotActivated,
} from './homepage.styles';

const HomePageThokimanContentContainer = lazy(() =>
  import(
    '../../components/homepage/homepage-contents/homepage-thokiman-container/homepage-thokiman-container.component'
  )
);
const HomePageProfileImageContainer = lazy(() =>
  import(
    '../../components/homepage/homepage-contents/homepage-profile-image-container/homepage-profile-image-container.component'
  )
);
const HomePageProfileTextContainer = lazy(() =>
  import(
    '../../components/homepage/homepage-contents/homepage-background-profile-text-container/homepage-background-profile-text-container.component'
  )
);

export const HomePage = (props) => (
  <HomepageContainer $issidebaractive={props.isSideBarActive}>
    {props.isSideBarActive ? (
      <HomepageIsNotActivated>
        Ho
        <br />
        me
      </HomepageIsNotActivated>
    ) : null}
    <Suspense fallback={<SpinnerLoading />}>
      <HomePageThokimanContentContainer />
      <HomepageAboutContentContainer>
        <HomePageProfileImageContainer />
        <HomePageProfileTextContainer {...props} />
      </HomepageAboutContentContainer>
    </Suspense>
  </HomepageContainer>
);

const mapStateToProps = createStructuredSelector({
  shortDescription: selectPersonalSummaryShortDescription,
  quote: selectPersonalSummaryQuote,
  isSideBarActive: selectIsSideBarActive,
});
export default connect(mapStateToProps)(HomePage);
