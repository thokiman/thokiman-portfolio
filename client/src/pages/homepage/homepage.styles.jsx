import styled, { css, keyframes } from 'styled-components';
const whiteSmokeColor = '#e3e1e4';
const maxWidthDesktop = 1366;
const maxWidthTabletPro = 1024;
const maxWidthSmartphone = 800;

export const HomepageElement = styled.div``;

const homepageSideBarHide = keyframes`
  0% {
    transform: rotateX(100deg);
    transform-origin: bottom;
    opacity: 0;
  }
  100% {
    transform: rotateX(0);
    transform-origin: bottom;
    opacity: 1;
  }
`;

const getTopHomepageIsNotActivatedProps = ({
  $homepageisnotactivatedprops,
}) => {
  return $homepageisnotactivatedprops.top;
};
const getLeftHomepageIsNotActivatedProps = ({
  $homepageisnotactivatedprops,
}) => {
  return $homepageisnotactivatedprops.left;
};
const getFontSizeHomepageIsNotActivatedProps = ({
  $homepageisnotactivatedprops,
}) => {
  return $homepageisnotactivatedprops.fontSize;
};

export const HomepageIsNotActivated = styled.div`
  font-family: spitfire;

  color: ${whiteSmokeColor};
  position: absolute;
  top: ${getTopHomepageIsNotActivatedProps}vh;
  left: ${getLeftHomepageIsNotActivatedProps}vw;
  font-size: ${getFontSizeHomepageIsNotActivatedProps}vw;
  letter-spacing: 0.17em;
  width: 50vw;
  line-height: 1.5em;

  animation: ${homepageSideBarHide} 2s cubic-bezier(0.175, 0.885, 0.32, 1.275)
    both;
`;

const HomepageIsActivated = css`
  @media only screen and (max-width: ${maxWidthDesktop}px) {
    position: relative;
    left: 0;
    width: 100vw;
    transition: left 0.5s ease;
    opacity: 1;
    height: 500vh;
  }
  @media only screen and (max-width: ${maxWidthTabletPro}px) {
    height: 380vh;
  }
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    height: 360vh;
  }
`;

const getHomepageSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return null;
  return HomepageIsActivated;
};
export const HomepageContainer = styled.div`
  text-align: center;
  transition: background-image 0.7s ease-in-out, height 0.4s ease-in-out;

  @media only screen and (max-width: ${maxWidthDesktop}px) {
    position: relative;
    top: 0;
    width: 64vw;
    left: 140px;
    right: 0;
    bottom: 0;
    transition: left 0.5s ease;
    overflow-y: hidden;
    overflow-x: hidden;
    opacity: 0.5;
    height: 70.8vh;
  }
  @media only screen and (max-width: 350px) {
    height: 71vh;
  }
  ${getHomepageSideBarActive};
`;

export const HomepageAboutContentContainer = styled.div`
  color: ${whiteSmokeColor};
  height: 120vh;
  transition: all 0.5s ease-in-out;

  display: flex;
  align-items: center;
  justify-content: space-between;
  @media only screen and (max-width: ${maxWidthDesktop}px) {
    flex-direction: column;
  }
`;

HomepageContainer.displayName = 'HomepageContainer';
HomepageAboutContentContainer.displayName = 'HomepageAboutContentContainer';
