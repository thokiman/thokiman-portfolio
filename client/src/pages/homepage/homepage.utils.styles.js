import {
  hasScreenSize1,
  hasScreenSize1_1,
  hasScreenSize1_2,
  hasScreenSize1_3,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize3_5,
  hasScreenSize4,
  hasScreenSize5,
  hasScreenSizeWindow_1,
} from 'utils/common.component.utils';

export const measureHomepageContainerProps = (viewWidth, viewHeight) => {
  const homepageOffset = 100;
  let homepageContainerProps;
  if (hasScreenSizeWindow_1(viewWidth, viewHeight)) {
    console.log('homepage', 1);
    homepageContainerProps = {
      height: 500 + homepageOffset,
    };
  } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
    console.log('homepage', 2);
    homepageContainerProps = {
      height: 400 + homepageOffset,
    };
  } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
    console.log('homepage', 3);
    homepageContainerProps = {
      height: 520 + homepageOffset,
    };
  } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
    console.log('homepage', 4);
    homepageContainerProps = {
      height: 520 + homepageOffset,
    };
  } else if (hasScreenSize1(viewWidth)) {
    console.log('homepage', 5);
    homepageContainerProps = {
      height: 400 + homepageOffset,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log('homepage', 6);
    homepageContainerProps = {
      height: 400 + homepageOffset,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log('homepage', 7);
    homepageContainerProps = {
      height: 400 + homepageOffset,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log('homepage', 8);
    homepageContainerProps = {
      height: 380 + homepageOffset,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log('homepage', 9);
    homepageContainerProps = {
      height: 400 + homepageOffset,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log('homepage', 10);
    homepageContainerProps = {
      height: 400 + homepageOffset,
    };
  } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
    console.log('homepage', 11);
    homepageContainerProps = {
      height: 360 + homepageOffset,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log('homepage', 12);
    homepageContainerProps = {
      height: 400 + homepageOffset,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log('homepage', 13);
    homepageContainerProps = {
      height: 380 + homepageOffset,
    };
  } else {
    console.log('homepage', 14);
    homepageContainerProps = {
      height: 440,
    };
  }
  return homepageContainerProps;
};

export const measureHomepageIsNotActivated = (viewWidth, viewHeight) => {
  let homeepageIsNotActivatedProps;

  if (hasScreenSize1(viewWidth)) {
    homeepageIsNotActivatedProps = {
      top: 10.5,
      left: -14,
      fontSize: 7,
    };
  } else if (hasScreenSize2(viewWidth)) {
    homeepageIsNotActivatedProps = {
      top: 13,
      left: -15,
      fontSize: 5,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    homeepageIsNotActivatedProps = {
      top: 12,
      left: -12,
      fontSize: 8,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    homeepageIsNotActivatedProps = {
      top: 12.5,
      left: -10,
      fontSize: 9,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    homeepageIsNotActivatedProps = {
      top: 12.5,
      left: -13,
      fontSize: 7,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    homeepageIsNotActivatedProps = {
      top: 14,
      left: -12,
      fontSize: 6,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    homeepageIsNotActivatedProps = {
      top: 15,
      left: -14,
      fontSize: 5,
    };
  } else if (hasScreenSize5(viewWidth)) {
    homeepageIsNotActivatedProps = {
      top: 13,
      left: -8,
      fontSize: 12,
    };
  } else {
    homeepageIsNotActivatedProps = {
      top: 10.5,
      left: -10,
      fontSize: 10,
    };
  }
  return homeepageIsNotActivatedProps;
};
