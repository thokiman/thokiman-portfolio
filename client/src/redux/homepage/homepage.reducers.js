import { other } from '../../data.object/other.data.js';

const INITIAL_STATE = {
  other,
};

const homepageReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    default:
      return state;
  }
};

export default homepageReducer;
