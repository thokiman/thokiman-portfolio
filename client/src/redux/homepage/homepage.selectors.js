import { createSelector } from 'reselect';

const selectAllHomepage = (state) => state.homepage;

export const selectRockWaveImage = createSelector(
  [selectAllHomepage],
  (homepage) => homepage.other.layout.items.rockWave
);
