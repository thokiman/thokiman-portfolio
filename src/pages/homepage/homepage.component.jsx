import React from "react";

import { about } from "../../data.object/abouts.data";
import { ReactComponent as BlueOutlineThokiman } from "../../assets/collections-icon/thokiman-icon/logo/outline/thokiman_head_logo_blue.svg";
import { ReactComponent as WhiteOutlineThokimanSkeleton } from "../../assets/collections-icon/thokiman-icon/page/thokiman_skeleton_white.svg";
import "./homepage.styles.scss";

const HomePage = () => {
  return (
    <div className="homepage-container">
      <div className="thokiman-content-container">
        <div className="thokiman-container">
          <div className="thokiman-head-container">
            <BlueOutlineThokiman className="thokiman-head-outline" />
          </div>

          <div className="thokiman-body-container">
            <WhiteOutlineThokimanSkeleton className="white-thokiman-skeleton" />
          </div>
        </div>
      </div>
      <div className="about-content-container">
        <div className="profile-image-container">
          <div className="background-profile-image">
            <div className="profile-image"></div>
          </div>
        </div>
        <div className="profile-text-container">
          <div className="background-profile-text">
            <div className="name-text">Thomas Santosa</div>
            <div className="activity-text">
              {about.aboutPath.items.personalSummary.items.shortDescription}
            </div>
            <div className="quote-text">
              {about.aboutPath.items.personalSummary.items.quote}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default HomePage;
