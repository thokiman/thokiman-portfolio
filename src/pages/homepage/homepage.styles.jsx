import styled, { css } from "styled-components";

const HomepageSideBarActive = css`
  @media only screen and (max-width: 800px) {
    position: relative;
    left: 0;
    width: 100vw;
    transition: left 0.5s ease;
    opacity: 1;
  }
`;
const getHomepageSideBarActive = (props) => {
  if (props.$issidebarhidden) return null;
  return HomepageSideBarActive;
};
export const HomepageContainer = styled.div`
  text-align: center;
  transition: background-image 0.7s ease-in-out, height 0.4s ease-in-out;

  @media only screen and (max-width: 800px) {
    position: relative;
    top: 0;
    width: 66vw;
    left: 140px;
    right: 0;
    bottom: 0;
    transition: left 0.5s ease;
    height: 115vh;
    overflow-y: scroll;
    overflow-x: hidden;
    opacity: 0.5;
  }
  ${getHomepageSideBarActive};
`;

export const HomepageAboutContentContainer = styled.div`
  color: #e3e1e4;
  height: 110vh;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

HomepageContainer.displayName = "HomepageContainer";
HomepageAboutContentContainer.displayName = "HomepageAboutContentContainer";
