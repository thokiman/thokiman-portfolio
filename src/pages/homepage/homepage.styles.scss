 .homepage-container {
     text-align: center;
     transition: background-image 0.7s ease-in-out, height 0.4s ease-in-out;

     .thokiman-content-container {
         display: flex;
         align-items: center;
         justify-content: center;
         height: 110vh;
         background-image: url('../../assets/layout/water-1.jpg');
         background-size: cover;

         .thokiman-container {
             height: 100%;
             width: 75%;
             display: flex;
             flex-direction: row;
             align-items: center;
             background-color: rgba(116, 198, 212, 0.5);

             .thokiman-head-container {

                 display: flex;
                 align-items: center;
                 justify-content: center;
                 height: 100%;
                 width: 40%;
                 position: relative;
                 left: 150px;


                 .thokiman-head-outline {
                     background-size: contain;
                     box-sizing: border-box;
                     width: 80%;
                     height: 100%;
                     background-color: #74c6d4;


                 }
             }

             .thokiman-body-container {

                 display: flex;
                 align-items: center;
                 justify-content: center;
                 height: 100%;
                 width: 60%;


                 .white-thokiman-skeleton {


                     width: 60%;
                     height: 100%;

                     .thokiman-skeleton-white {
                         stroke-miterlimit: 10;
                         stroke-dasharray: 700 150;
                         stroke-dashoffset: 100%;
                         animation: thokiman-skeleton-white 2s ease-in-out alternate-reverse infinite;

                         @keyframes thokiman-skeleton-white {
                             to {
                                 stroke-dashoffset: 0%;
                             }
                         }
                     }




                 }
             }
         }

     }

     .about-content-container {

         color: white;
         height: 110vh;
         display: flex;
         align-items: center;
         justify-content: space-between;



         .profile-image-container {

             height: 100%;
             width: 42%;
             display: flex;
             align-items: center;
             justify-content: center;
             position: relative;
             background-color: #080808;


             .background-profile-image {

                 width: 80%;
                 height: 90%;
                 display: flex;
                 justify-content: center;
                 align-items: center;

                 background-size: cover;

                 .profile-image {

                     width: 97%;
                     height: 97%;
                     background-image: url('../../assets/profile/profile.png');
                     background-size: cover;
                 }
             }
         }

         .profile-text-container {
             height: 100%;
             width: 55%;
             display: flex;
             align-items: center;
             justify-content: center;
             position: relative;
             right: 60px;



             .background-profile-text {

                 width: 95%;
                 height: 75%;
                 display: flex;
                 flex-direction: column;
                 align-items: center;
                 justify-content: center;


                 .name-text {
                     text-align: center;
                     font-kerning: auto;
                     font-family: angilla;
                     letter-spacing: 0.005em;
                     font-size: 930%;
                     width: 100%;
                     height: 23%;
                     word-spacing: -35px;
                 }

                 .activity-text {

                     color: #9096ae;
                     font-family: baskerville-old-face;
                     font-size: 240%;
                     font-kerning: auto;
                     letter-spacing: 0.03em;
                     text-align: center;
                     line-height: 1.1em;
                     width: 100%;
                     height: 13%;

                 }

                 .quote-text {

                     color: #9096ae;
                     font-family: baskerville-old-face;
                     font-size: 150%;
                     font-kerning: auto;
                     letter-spacing: 0.01em;
                     text-align: center;
                     line-height: 1.15em;
                     height: 35%;
                     width: 90%;
                     height: 12%;
                 }
             }
         }
     }

 }