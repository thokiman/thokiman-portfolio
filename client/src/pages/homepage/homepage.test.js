import React from "react";
import { shallow } from "enzyme";

import { HomePage } from "./homepage.component";

describe("Homepage page", () => {
  let wrapper;
  const mockText = "";
  const mockFalseBool = false;
  beforeEach(() => {
    const mockProps = {
      shortDescription: mockText,
      quote: mockText,
      isSideBarHidden: mockFalseBool,
    };
    wrapper = shallow(<HomePage {...mockProps} />);
  });
  it("should render HomePage page", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should not render HomepageSideBarHide if isSideBarHidden is false", () => {
    expect(wrapper.exists("HomepageSideBarHide")).toBe(false);
  });
});
