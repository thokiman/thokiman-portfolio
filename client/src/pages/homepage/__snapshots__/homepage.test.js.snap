// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage page should render HomePage page 1`] = `ShallowWrapper {}`;
